src/asar.ts

```diff
@@ -0,0 +1,265 @@
+import * as path from 'path';
+import * as minimatch from 'minimatch';
+
+import fs from './wrapped-fs';
+import { Filesystem, FilesystemEntry } from './filesystem';
+import * as disk from './disk';
+import { crawl as crawlFilesystem, determineFileType } from './crawlfs';
+import { IOptions } from 'glob';
+
+/**
+ * Whether a directory should be excluded from packing due to the `--unpack-dir" option.
+ *
+ * @param {string} dirPath - directory path to check
+ * @param {string} pattern - literal prefix [for backward compatibility] or glob pattern
+ * @param {array} unpackDirs - Array of directory paths previously marked as unpacked
+ */
```

Comment:
```suggestion
/**
 * Whether a directory should be excluded from packing due to the `--unpack-dir" option.
 *
 * @param dirPath - directory path to check
 * @param pattern - literal prefix [for backward compatibility] or glob pattern
 * @param unpackDirs - Array of directory paths previously marked as unpacked
 */
```

We should use TSDoc comments rather than JSDoc style after the conversion. The parameter types are redundant and often go stale.